src/components/Logout/index.tsx

```diff
@@ -0,0 +1,27 @@
+import { EXPRESS_OAUTH_LOGOUT_URL } from '../../configs/env';
+
+/** interface to describe props for Logout component
+ * @member {string} logoutURL the url of the logout endpoint of the Oauth server.
+ */
+export interface LogoutProps {
+  logoutURL: string;
+}
+
+export function logoutFromAuthServer(logoutURL: string) {
+  const logoutWindow: Window | null = window.open(logoutURL);
+  const timer: NodeJS.Timeout = setInterval(() => {
+    if (logoutWindow) {
+      logoutWindow.close();
+      window.location.href = EXPRESS_OAUTH_LOGOUT_URL;
+    }
+    clearInterval(timer);
+  }, 20);
+}
+
+/** Logout component */
+const Logout = (props: LogoutProps) => {
+  logoutFromAuthServer(props.logoutURL);
+  return null;
+};
```

Comment:
If we're doing this via a simple functional component, could / should this component return a `<Redirect />` component (instead of null) so we don't have to manually set `window.location.href`? 🤔

Reply:
Cool will give this a shot

Reply:
@kahummer @cKellyDesign the logout action here needs to be served by express server. Using Redirect will limit the logout to the react-apps' scope and the express-server will never know of it.

Reply:
@p-netm I've actually confirmed this

Reply:
cool